CHANGELOG.md

```diff
@@ -1,3 +1,68 @@
+## 2022-11-14 v0.13.0
+
+Release highlights:
+- A [Waku canary tool](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/tools#waku-canary-tool) to check if nodes are reachable and what protocols they support.
+- Simplified configuration for store protocol. This [new guide](https://github.com/status-im/nwaku/blob/4e5318bfbb204bd1239c95472d7b84b6a326dd9d/docs/operators/how-to/configure-store.md) explains how to configure store from this release forward.
+- Support for environment variables to configure a nwaku node. See our [configuration guide](https://github.com/status-im/nwaku/blob/384abed614050bf3aa90c901d7f5e8bc383e8b22/docs/operators/how-to/configure.md) for more.
+- A Waku [network monitoring tool](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/tools#networkmonitor) to report network metrics, including network size, discoverable peer capabilities and more.
+
+### Features
+
+- Added Waku canary tool to check if i) a given node is reachable and ii) it supports a set of protocols.
+- Simplified [Waku store configuration](https://github.com/status-im/nwaku/blob/4e5318bfbb204bd1239c95472d7b84b6a326dd9d/docs/operators/how-to/configure-store.md).
+- Decoupled Waku peer persistence configuration from message store configuration.
+- Added keyfile support for secure storage of RLN credentials.
+- Added configurable libp2p agent string to nwaku switch.
+- Support for [configuration with environment variables](https://github.com/status-im/nwaku/blob/384abed614050bf3aa90c901d7f5e8bc383e8b22/docs/operators/how-to/configure.md).
+- Added [example module](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/examples/v2) to showcase basic nwaku relay usage.
+- Added a nwaku [network monitoring tool](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/tools#networkmonitor) to provide metrics on peers, network size and more.
+
+### Changes
+
+- Improved logging for [`17/WAKU-RLN-RELAY`](https://rfc.vac.dev/spec/17/) implementation.
+- Connection to eth node for RLN now more stable, maintains state and logs failures.
+- Waku apps and tools now moved to their own subdirectory.
+- Continued refactoring of several protocol implementations to improve maintainability and readability.
+- Periodically log metrics when running RLN spam protection.
+- Added metrics dashboard for RLN spam protection.
+- Github CI test workflows are now run selectively, based on the content of a PR.
+- Improved reliability of CI runs and added email notifications.
+- Discv5 discovery loop now triggered to fill a [34/WAKU2-PEER-EXCHANGE](https://rfc.vac.dev/spec/34/) peer list cache asynchronously.
+- Upgraded to Nim v1.6.6.
+- Cleaned up compiler warnings on unused imports.
+- Improved exception handling and annotation.
+
+### Fixes
```

Comment:
Personal opinion, but I like having the PR as part of the release notes, specially for the fixes. wdyt?
As a reference I like https://github.com/prysmaticlabs/prysm/releases a lot :)

Reply:
> I like having the PR as part of the release notes

I agree! Since automated release notes are not a thing yet and this is a bit manual, I've just added the PRs now to the fixes. [3cf5cb5](https://github.com/status-im/nwaku/pull/1378/commits/3cf5cb59e005e93dacd01ec4ccb846e4433f4a26)

Reply:
I usually automatically generate the notes with this. You get all changes + linked PR. And then you just have to make some minor modifications to adapt it. The autogenerated text is quite complete. not sure if we must include every single PR, but it's useful as a reference.

<img width="290" alt="image" src="https://user-images.githubusercontent.com/8811422/201732636-2dc43788-6acd-49b8-b4f9-07c30548e691.png">

## What's Changed
* chore(rln-relay): adds Rln-relay testnet2 tutorial by @staheri14 in https://github.com/status-im/nwaku/pull/1232
* chore(rln-relay): disable merkle root validation by @rymnc in https://github.com/status-im/nwaku/pull/1246
* feat(RLN): Add verify against multiple roots by @s1fr0 in https://github.com/status-im/nwaku/pull/1250
* chore(rln-relay): additional logging by @rymnc in https://github.com/status-im/nwaku/pull/1243
* chore(rln): update zerokit module by @s1fr0 in https://github.com/status-im/nwaku/pull/1254
* test(rln-relay): setup heartbeat ws subscription and log when connection drops by @rymnc in https://github.com/status-im/nwaku/pull/1253
* docs: fix broken Kibana link by @jm-clius in https://github.com/status-im/nwaku/pull/1255
* fix(rln): change IdCommitment encoding to LE by @s1fr0 in https://github.com/status-im/nwaku/pull/1256
* fix(rln): set maxEpochGap as maximum epoch gap by @s1fr0 in https://github.com/status-im/nwaku/pull/1257
* chore(wakucanary): add canary tool by @alrevuelta in https://github.com/status-im/nwaku/pull/1205
* chore(rln-relay): make account address optional by @rymnc in https://github.com/status-im/nwaku/pull/1258
* refactor(rln): display byte arrays in hex as uints and fix Kilic's proofVerify proc by @s1fr0 in https://github.com/status-im/nwaku/pull/1259
* chore: move apps and tools to their own directory by @LNSD in https://github.com/status-im/nwaku/pull/1228
* ci: add username and email to build notifications by @jakubgs in https://github.com/status-im/nwaku/pull/1260
* feat(rln-relay): periodically log metrics by @rymnc in https://github.com/status-im/nwaku/pull/1262
* refactor(wakunode2): move config random private key generation to a procedure by @LNSD in https://github.com/status-im/nwaku/pull/1273
* refactor(node): split wakunode into waku_node and wakunode2 by @LNSD in https://github.com/status-im/nwaku/pull/1274
* chore(wakucanary): resolve dns by @alrevuelta in https://github.com/status-im/nwaku/pull/1271
* chore(wakunode2): move wakunode2 to apps directory by @LNSD in https://github.com/status-im/nwaku/pull/1277
* fix: log-level being ignored by @alrevuelta in https://github.com/status-im/nwaku/pull/1272
* feat(rln-relay): add metrics dashboard by @rymnc in https://github.com/status-im/nwaku/pull/1275
* fix: store cursors should be exclusive and match a DB item by @jm-clius in https://github.com/status-im/nwaku/pull/1263
* chore(store): disable temporarily the store resume flaky test case by @LNSD in https://github.com/status-im/nwaku/pull/1281
* feat(store): add waku store client module by @LNSD in https://github.com/status-im/nwaku/pull/1229
* chore(rln-relay): Changes the test scenario of testnet2 tutorial to one chat user by @staheri14 in https://github.com/status-im/nwaku/pull/1261
* ci: run ci test workflows selectively by @LNSD in https://github.com/status-im/nwaku/pull/1284
* feat(chat2): metrics server + metrics logging by @rymnc in https://github.com/status-im/nwaku/pull/1279
* refactor(wakunode2): extract setup proc definitions from main section by @LNSD in https://github.com/status-im/nwaku/pull/1286
* chore: extract test procedures and constants into a common module by @LNSD in https://github.com/status-im/nwaku/pull/1287
* docs: add docker and quickstart guides by @jm-clius in https://github.com/status-im/nwaku/pull/1289
* fix: call runPeerExchangeDiscv5Loop by @kaiserd in https://github.com/status-im/nwaku/pull/1297
* feat(lightpush): add waku lightpush protocol client by @LNSD in https://github.com/status-im/nwaku/pull/1295
* ci(jenkins): use MAKEFLAGS and NIMFLAGS pipeline params in build and test stages by @LNSD in https://github.com/status-im/nwaku/pull/1299
* fix(wakunode2): no peer-exchange peer error msg by @kaiserd in https://github.com/status-im/nwaku/pull/1298
* feat(store): simplify wakunode2 configuration options by @LNSD in https://github.com/status-im/nwaku/pull/1293
* chore(node): waku node code reorganization by @LNSD in https://github.com/status-im/nwaku/pull/1303
* feat(wakunode2): simplify wakunode2 config and decouple peer persistence by @LNSD in https://github.com/status-im/nwaku/pull/1300
* chore(rln-relay): removes the Goerli account address config option from the onchain-rln-relay tutorial by @staheri14 in https://github.com/status-im/nwaku/pull/1280
* feat: add keyfile support for RLN credentials secure storage by @s1fr0 in https://github.com/status-im/nwaku/pull/1285
* chore: update submodules and bump Nim to 1.6.6 by @jm-clius in https://github.com/status-im/nwaku/pull/1307
* feat(testlib): add enable/disable test debug utils by @LNSD in https://github.com/status-im/nwaku/pull/1315
* feat(wakunode2): add nwaku agent-string to switch by @alrevuelta in https://github.com/status-im/nwaku/pull/1302
* refactor(node): use lightpush client in waku_node by @LNSD in https://github.com/status-im/nwaku/pull/1308
* chore: remove legacy unused test suite by @LNSD in https://github.com/status-im/nwaku/pull/1316
* refactor(node): use store client in waku_node by @LNSD in https://github.com/status-im/nwaku/pull/1309
* chore(common): move sqlite module to common module by @LNSD in https://github.com/status-im/nwaku/pull/1317
* docs(operator-trial): add rln instructions by @rymnc in https://github.com/status-im/nwaku/pull/1311
* fix(rln-relay): RangeDefect test by @rymnc in https://github.com/status-im/nwaku/pull/1318
* feat(rln-relay): track last seen event by @rymnc in https://github.com/status-im/nwaku/pull/1296
* refactor(peer-exchange): move peer management to waku_node module by @LNSD in https://github.com/status-im/nwaku/pull/1320
* feat(filter): split waku filter protocol into server and client by @LNSD in https://github.com/status-im/nwaku/pull/1319
* fix: replace TaintedString type deprecated in nim 1.5 by @LNSD in https://github.com/status-im/nwaku/pull/1326
* fix: replace nim's std/nre with nim-regex library by @LNSD in https://github.com/status-im/nwaku/pull/1327
* refactor(sqlite): move migrations logic to sqlite common module by @LNSD in https://github.com/status-im/nwaku/pull/1325
* fix(wakunode2): fix message retention policy config validation regex by @LNSD in https://github.com/status-im/nwaku/pull/1328
* refactor(wakunode2): move configuration load logic to config module by @LNSD in https://github.com/status-im/nwaku/pull/1329
* chore: randomize waku tcp ports in tests by @LNSD in https://github.com/status-im/nwaku/pull/1336
* chore: make log topics consistent with nim-chronicles style by @LNSD in https://github.com/status-im/nwaku/pull/1334
* chore(migrations): move migration scripts to project root by @LNSD in https://github.com/status-im/nwaku/pull/1330
* feat(testlib): mark commented test cases as skipped by @LNSD in https://github.com/status-im/nwaku/pull/1337
* feat(wakunode2): support configuration via environment variables by @LNSD in https://github.com/status-im/nwaku/pull/1331
* chore(rln-relay): remove all raises and replace with Result types by @rymnc in https://github.com/status-im/nwaku/pull/1321
* refactor(peer_manager): move peer_store under peer_manager module by @LNSD in https://github.com/status-im/nwaku/pull/1339
* chore: remove unused imports by @LNSD in https://github.com/status-im/nwaku/pull/1340
* chore: update push raises pragma according to nim 1.6 by @LNSD in https://github.com/status-im/nwaku/pull/1341
* ci: fix path to migrations in Dockerfile by @jakubgs in https://github.com/status-im/nwaku/pull/1343
* chore(examples): add pubsub example with production env by @alrevuelta in https://github.com/status-im/nwaku/pull/1333
* refactor(message_store): move message_store to node module by @LNSD in https://github.com/status-im/nwaku/pull/1342
* chore(wakucanary): remove port hardcoding + add cli config by @alrevuelta in https://github.com/status-im/nwaku/pull/1345
* chore: build rln dependecies only when building v2 by @LNSD in https://github.com/status-im/nwaku/pull/1344
* chore: rename protocols rpc_codec procs from init to decode by @LNSD in https://github.com/status-im/nwaku/pull/1346
* fix(rln-relay): remove dependency on applications' configuration by @LNSD in https://github.com/status-im/nwaku/pull/1351
* chore(rln): update ganache by @s1fr0 in https://github.com/status-im/nwaku/pull/1347
* refactor(queue_store): rename queue_store module and simplify api by @LNSD in https://github.com/status-im/nwaku/pull/1348
* docs: extend docs with the different node configuration mechanisms by @LNSD in https://github.com/status-im/nwaku/pull/1355
* refactor: unify pubsub topic type and default value by @LNSD in https://github.com/status-im/nwaku/pull/1352
* refactor: continue pubsub/content types started in #1352 by @alrevuelta in https://github.com/status-im/nwaku/pull/1362
* refactor(store): decouple waku store public api types from rpc types by @LNSD in https://github.com/status-im/nwaku/pull/1350
* refactor(rln): remove kilic lib by @s1fr0 in https://github.com/status-im/nwaku/pull/1365
* chore(networkmonitor): tool to discover and provide metrics on peers by @alrevuelta in https://github.com/status-im/nwaku/pull/1290
* chore: upgrade nim-stew to latest version by @LNSD in https://github.com/status-im/nwaku/pull/1366
* docs: extend waku store config tutorial of the operator docs by @LNSD in https://github.com/status-im/nwaku/pull/1359
* feat(rln-relay): process blocks atomically by @rymnc in https://github.com/status-im/nwaku/pull/1349
* chore(networkmonitor): add metric listing content topics + messages by @alrevuelta in https://github.com/status-im/nwaku/pull/1335
* chore: disable store protocol by default by @jm-clius in https://github.com/status-im/nwaku/pull/1374
* fix(rln-relay): feature guard by @rymnc in https://github.com/status-im/nwaku/pull/1373

## New Contributors
* @alrevuelta made their first contribution in https://github.com/status-im/nwaku/pull/1205

**Full Changelog**: https://github.com/status-im/nwaku/compare/v0.12.0...v0.13.0

Reply:
This looks really nice ^!

Reply:
Indeed! Let's consider for next release. May want to introduce more sections and remove some confusing repeated items, but I like the approach.